src/rest_api/classes/feature/widgets/associations.clj

```diff
@@ -0,0 +1,113 @@
+(ns rest-api.classes.feature.widgets.associations
+  (:require
+    [pseudoace.utils :as pace-utils]
+    [rest-api.formatters.object :as obj :refer  [pack-obj]]
+    [rest-api.classes.generic-fields :as generic]))
+
+(defn associations [f]
+  {:data (not-empty
+           (sort-by :label
+                    (flatten
+                      (vals
+                        (pace-utils/vmap
+                          :cds
+                          (some->> (:feature/associated-with-cds f)
+                                   (map :feature.associated-with-cds/cds)
+                                   (map #(pace-utils/vmap
+                                           :association (pack-obj %)
+                                           :label "CDS")))
+
+                          :feature
+                          (some->> (:feature/associated-with-feature f)
+                                   (map :feature.associated-with-feature/feature)
+                                   (map #(pace-utils/vmap
+                                           :association (pack-obj %)
+                                           :label "Feature")))
+
+                          :gene
+                          (some->> (:feature/associated-with-gene f)
+                                   (map :feature.associated-with-gene/gene)
+                                   (map #(pace-utils/vmap
+                                           :association (pack-obj %)
+                                           :label "Gene")))
+
+                          :operon
+                          (some->> (:feature/associated-with-operon f)
+                                   (map :feature.associated-with-operon/operon)
+                                   (map #(pace-utils/vmap
```

Comment:
Maybe:
```
(map #(hash-map :association (pack-obj %)
                              :label "Operon"))
```
instead of using `pace-utils/vmap`.
Generally `vmap` is only useful if values could be nil (in which case key/val pairs won't be put in the resultant map).

Reply:
fixed with next commit